README.adoc

```diff
@@ -1,88 +1,323 @@
-* https://github.com/jboss-developer/jboss-developer-shared-resources/blob/master/guides/CONTRIBUTING.adoc#jboss_developer_contributing_guide[Contributing Guide]: This document contains information targeted for developers who want to contribute to {productName} developer projects.
-* xref:available_quickstarts[Available Quickstarts]: List of the available quickstarts and details about each one.
+[[system_requirements]]
+== System Requirements
 
-[[use_of_product_home_and_jboss_home_variables]]
-== Use of {jbossHomeName} and JBOSS_HOME Variables
+The applications these projects produce are designed to be run on WildFly 16 or later.
 
-The quickstart `README` files use the _replaceable_ value `__{jbossHomeName}__` to denote the path to the {productName} installation. When you encounter this value in a `README` file, make sure you replace it with the actual path to your {productName} installation. The installation path is described in detail here: link:{useProductHomeDocUrl}[Use of __{jbossHomeName}__ and __JBOSS_HOME__ Variables].
+All you need to build these projects is Java 8.0 (Java SDK 1.8) or later and Maven 3.3.1 or later.
 
-// Suggested Approach to the Quickstarts
-include::shared-doc/suggested-approach.adoc[leveloffset=+1]
+[[prerequisites]]
+== Prerequisites
 
-[[system_requirements]]
-== System Requirements
+Before you begin, you must perform the following tasks.
 
-The applications these projects produce are designed to be run on {productNameFull} {productVersion} or later.
+. xref:build_wildfly_server_and_boms[Build the WildFly server and BOMs]: Fetch the latest WildFly source code and build the WildFly Server and the WildFly BOMs. This installs the Maven artifacts that are required by the WildFly quickstarts.
+
+. xref:install_the_quickstart_parent_artifact_in_maven[Install the Quickstart Parent Artifact in Maven]: The root `POM.xml` file defines dependencies that are required by some of the quickstarts.
+
+. xref:build_quickstart_readme_files[Build the Quickstart README.html Files]: The quickstart `README` files are written in AsciiDoc to provide modular, reusable content; however, this makes them difficult to read. For this reason, you also must build the quickstart `README.html` files from the AsciiDoc source.
+
+[[build_wildfly_server_and_boms]]
+=== Build the WildFly Server and BOMs
+
+You must build the WildFly server and BOMs to install the Maven artifacts before you can build a quickstart or a quickstart's `README.html` file. The WildFly development team normally takes care of building and publishing the Maven artifacts needed to build the quickstarts in this repository, but as with all new development, there might be times where you need a library that has not yet been updated. Occasionally you might also need an update that was made to WildFly core.
+
+The order for building the WildFly server is as follows.
+
+. xref:build_wildfly_core[Build WildFly core] (if required)
```

Comment:
I include this because at one time, I couldn't run a new quickstart without building core. But you are right, it is very rare and it complicates things. I'll remove it.